### PR TITLE
fix(es/loader): Make `tsc` resolver work for bare specifier

### PR DESCRIPTION
**Pull request by XHMM**
**Description:**
From the typescript [baseUrl](https://www.typescriptlang.org/docs/handbook/modules/reference.html#baseurl) doc: When using bare specifiers (module specifiers that don’t begin with ./, ../, or /), baseUrl has a higher precedence than node_modules package lookups. 

In the current tsc resolver implementation, when resolving bare module specifiers, baseUrl was not used except for `paths`, this cause the belowing resolution failed,  but it worked when used in typescript project.

`tsconfig.json` / `.swcrc`:
```json
{
    baseUrl: "."
    paths: {
        "@common/*": ["src/common/*"]
    }
}
```

File structure:
- ./src/common/helper.ts
- ./src/index.ts

./src/index.ts content:
```ts
 // tsc can resolve this, but tsc resolver cannot
import sth from "src/common/helper"
```

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

File: crates/swc_ecma_loader/src/resolvers/tsc.rs [hunk 1]
```diff
@@ -310,6 +310,13 @@ where
...
         }
 
+        // https://www.typescriptlang.org/docs/handbook/modules/reference.html#baseurl
+        if let Ok(v) =
+            self.invoke_inner_resolver(base, self.base_url.join(module_specifier).to_str().unwrap())
+        {
+            return Ok(v);
+        }
+
         if let Ok(v) = self.invoke_inner_resolver(&self.base_url_filename, module_specifier) {
```
Comment (kdy1) [inline crates/swc_ecma_loader/src/resolvers/tsc.rs:320]: Please remove this line

Comment (XHMM) [reply crates/swc_ecma_loader/src/resolvers/tsc.rs]: Done

**Review by kdy1 — APPROVED** [review]
Thanks!

---


swc-bump:
 - swc_ecma_loader

**Review by kdy1 — CHANGES_REQUESTED** [review]
CI failed

File: crates/swc_ecma_loader/tests/tsc_resolver.rs [hunk 2]
```diff
@@ -97,6 +97,46 @@ fn pattern_1() {
     }
 }
 
+#[test]
+fn base_url_precedence() {
```
Comment (kdy1) [inline crates/swc_ecma_loader/tests/tsc_resolver.rs:101]: The test name says that it's a test about precedence, but it's not testing precedence. Can you add some files to https://github.com/swc-project/swc/tree/039c1685c663cd138e4a0cf769b506d1fd419ba5/crates/swc_ecma_loader/tests and check if it's actually precedence?

Comment (XHMM) [reply crates/swc_ecma_loader/tests/tsc_resolver.rs]: I'vs updated test cases into two cases for more clear. 😊

**Review by kdy1 — APPROVED** [review]
Thank you!

---

swc-bump: 
 - swc_ecma_loader

**Pull request merged**